src/odoc/interface.mld

```diff
@@ -0,0 +1,95 @@
+{0 Odoc interface guarantees}
+
+Odoc has several 'public facing' parts, with varying levels of support guarantees.
+This document describes what those interfaces are and what the support levels are
+now and what we aim for in the future.
+
+{2 Documentation comments}
+
+The first and most important is the syntax of the documentation comments present in source code.
+This is relevant to everyone who is writing code intended to be documented by odoc, and hence is applies to the widest set of people.
+
+The canonical description of the markup that odoc understands is in {{:https://caml.inria.fr/pub/docs/manual-ocaml/ocamldoc.html#s%3Aocamldoc-comments}this section}
+of the OCaml reference manual. The eventual aim is to support the in-code markup
+in its entirety, although right now there are some gaps. There are also some
+extensions where odoc goes beyond what is officially supported.
+
+{3 Changes}
+
+The following describes the changes between what odoc understands and what is in the OCaml manual.
+
+{4 Omissions}
+- Comments describing class inheritance are not rendered (bug)
+- Odoc does not currently handle ambiguously associated doc comments (bug)
+- Odoc does not ignore tags where they don't make sense (e.g. [@param] tags on instance variables are rendered) (bug)
+- {{:https://caml.inria.fr/pub/docs/manual-ocaml/ocamldoc.html#ss:ocamldoc-formatting}Alignment elements} are not handled ([{C text}], [{L text}] and [{R text}])
+- Odoc does not recognise {{:https://caml.inria.fr/pub/docs/manual-ocaml/ocamldoc.html#sss:ocamldoc-html-tags}html tags embedded in comments}
+
+{4 Improvements}
+- Odoc has a better mechanism for disambiguating references in comments. See 'reference syntax' later in this document.
+- Support for standalone 'mld' files - these are documents using the OCamldoc markup, but rendered as distinct pages.
+- A few extra tags are supported:
+  + [@returns] is a synonym for [@return]
+  + [@raises] is a synonym for [@raise]
+  + [@open] and [@closed] and [@inline] are hints for how 'included' signatures should be rendered
+  + [@canonical] allows a definition of a module to be marked as canonically elsewhere. 
```

Comment:
@lpw25 from dev meeting: we dont support !modules and the list of modules, and the table of contents is also unsupported. These are omissions.

Reply:
Except or the [lack of synopsis](https://github.com/ocaml/odoc/issues/297) `!modules` is perfectly supported. I use it in almost every of my packages out there and in `odig`'s automatic `index.mld` generation for packages that do not provide one.

Reply:
I've added `{!indexlist}` in the omissions and added the issue @dbuenzli mentioned to the 2.0 milestone.